app-experimental.tex

```diff
@@ -2395,3 +2395,219 @@ \section{Distributed Capabilities For Peripherals And Accelerators} % <<<
+One approach is to allow unsealing within a CID space.
+This allows all compartments in that space to use that capability, but no other compartment can unseal the capability.
+This protects from transitive capability leaks outside of the CID space.
+As an addition, we envision a bit that -- if set -- allows sharing within the CID space.
+\end{itemize}
+
+Furthermore, we also envision the possibility of \textit{resealing} capabilities.
+A capability belonging to one compartment can be resealed to another compartment.
+This effectively means that code can seal one or more of its own capabilities for another compartment.
+The main advantages are that this avoids capabilities to be unsealed in the open as done with explicit unsealing as well as potential performance improvements due not needing to unseal and seal capabilites.
+However, resealing brings the disavantage for the receiving compartment of not knowing whether a capability was sealed by itself or another compartment.
+Therefore, we would expect the need for additional validation checks.
+
+\subsection{Explicit Unsealing}
+
+We need to reserve a CID value that represents an unsealed capability instead of a valid compartment ID.
+We can choose any value to represent an unsealed capability.
+In this proposal, a capability is unsealed if and only if  its ACID is set to 0.
+This ACID is also known as the zero ACID. A capability with the zero ACID can be CID sealed by any compartment.
+This introduces two new instructions (in comparison to the three-operand operations currently present in the CHERI-RISC-V, our proposed operations have two operands. The third implicit operand is PCC):
+
+\texttt{CCIDUnseal cd, cs1}\\
+If cs1.CID and PCC.CID match, then cs1 will be assigned to cd with cd.CID=0. Otherwise, throw an exception.
+
+Preliminary encoding(31:0): 0x7f, 0x1a, rs0, 0x0, cd, 0x5b (assign a random free funct5)
+
+\texttt{CCIDSeal cd, cs1}\\
+If cs1.CID==0, then cs1 will be assigned cd with cd.CID=PCC.CID. Otherwise, throw an exception.
+
+Preliminary encoding(31:0): 0x7f, 0x1b, rs0, 0x0, cd, 0x5b (assign a random free funct5)
+
+A capability with CID==0 can be used by any compartment.
+An unsealed capability can be misused and the system can become victim to transitive capability leaks, e.g., by code that carelessly stores capabilities in shared memory.
+
+Having the zero CID represent an unsealed capability seems untuitive and allows code unaware of CID sealing to operate correct.
+
+\subsection{CID Spaces}
+\label{subsec:cid_spaces}
+
+We envision that CIDs can build subspaces in order to model relationships between compartments.
+One possibility is to put compartments with identical upper bits into the same group, e.g., CID\_Group = 0b10101010xx.
+This would lead to a CID group of size 4.
+
+CID spaces can express trust relationships between compartments, e.g., within a CID space compartments can implicitly unseal capabilities of other compartments of the same CID space.
```

Comment:
>within a CID space compartments can implicitly unseal capabilities of other compartments of the same CID space.

-> "capabilities within a CID space are implicitly unsealed in any compartment of that CID space."